pkg/nsx/services/subnet/builder.go

```diff
@@ -36,11 +36,11 @@ func (service *SubnetService) buildSubnetSetID(subnetset *v1alpha1.SubnetSet, in
 }
 
 func (service *SubnetService) buildSubnetName(subnet *v1alpha1.Subnet) string {
-	return util.GenerateTruncName(common.MaxSubnetNameLength, subnet.ObjectMeta.Name, SUBNETPREFIX, "", "", getCluster(service))
+	return util.GenerateTruncName(common.MaxSubnetNameLength, subnet.ObjectMeta.Name, subnet.UID, SUBNETPREFIX, "", "", getCluster(service))
 }
 
 func (service *SubnetService) buildSubnetSetName(subnetset *v1alpha1.SubnetSet, index string) string {
-	return util.GenerateTruncName(common.MaxSubnetNameLength, subnetset.ObjectMeta.Name, SUBNETPREFIX, index, "", getCluster(service))
+	return util.GenerateTruncName(common.MaxSubnetNameLength, subnetset.ObjectMeta.Name, subnetset.UID, SUBNETPREFIX, index, "", getCluster(service))
```

Comment:
Here the index is a uuid string. Previously the subnet name for subnetset looks like:
```
  "display_name" : "sub-b720ee2c-5788-4680-9796-0f93db-6bba7552-f0e19e9d-2078-4f57-8491-92224def3561",
  "tags" : [ {
    "scope" : "nsx-op/cluster",
    "tag" : "b720ee2c-5788-4680-9796-0f93db33d8a9"
  }, {
    "scope" : "nsx-op/version",
    "tag" : "1.0.0"
  }, {
    "scope" : "nsx-op/subnetset_name",
    "tag" : "pod-default"
  }, {
    "scope" : "nsx-op/subnetset_uid",
    "tag" : "28e85c0b-21e4-4cab-b1c3-597639dfe752"
  }, {
    "scope" : "nsx-op/namespace_uid",
    "tag" : "ed5cf5ad-c9b7-4c93-b4c8-d9fe1ef1ea96"
  }, {
    "scope" : "nsx-op/namespace",
    "tag" : "qe"
  }, {
    "scope" : "kubernetes.io/metadata.name",    "tag" : "qe"
  }, {
    "scope" : "vSphereClusterID",    "tag" : "domain-c10"
  }, {
    "scope" : "_abstractionPath_",
    "tag" : "/orgs/default/projects/project-quality/vpcs/ed5cf5ad-c9b7-4c93-b4c8-d9fe1ef1ea96/subnets/sub_28e85c0b-21e4-4cab-b1c3-597639dfe752_f0e19e9d-2078-4f57-8491-92224def3561"
  } ]
```
In your attachment (I prefer to the plain text next time because I counted the characters one by one manually in the image...) https://private-user-images.githubusercontent.com/5153872/350963774-a9022578-2c5b-40f5-9b01-07541bbc13cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4MDU4NjYsIm5iZiI6MTcyMTgwNTU2NiwicGF0aCI6Ii81MTUzODcyLzM1MDk2Mzc3NC1hOTAyMjU3OC0yYzViLTQwZjUtOWIwMS0wNzU0MWJiYzEzY2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjRUMDcxOTI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGIxYThiYjRjYzI4MTVjNWJiYjQ5Y2RhMTc3Y2U4MThhMTdhMmZkODE2MGRlMmQ2NmFlMjViYjU2OTY0YjBiNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.iYRitX74ptlwSUI5obUXFdMB2K8xENoO0Deo5FlesvQ, it seems that the index is lost, is it expected?